src/include/uapi/ipc/info.h

```diff
@@ -79,16 +79,17 @@ struct sof_ipc_fw_ready {
 	uint32_t hostbox_size;
 	struct sof_ipc_fw_version version;
 
-	/* Miscellaneous debug flags showing build/debug features enabled */
+	/* Miscellaneous flags showing build/debug features enabled */
 	union {
 		uint64_t reserved;
 		struct {
 			uint64_t build:1;
 			uint64_t locks:1;
 			uint64_t locks_verbose:1;
 			uint64_t gdb:1;
+			uint64_t no_xrun_recovery:1;
 		} bits;
-	} debug;
+	} flags;
```

Comment:
@slawblauciak  this name change will need driver update too, not sure if it's too late atm @plbossart  ? We can could apply to FW without the name change for the moment

Reply:
@slawblauciak do you have a small PR for the kernel yet with the same change ? Should be trivial to merge them both together.

Reply:
is there a clear behavior on the position of a new bitfield? will this result in this field being added on top, or all fields being moved by one, which is a non-compatible move. A compatible move would be to declare a new structure with the new field in the union.

Reply:
You're right. The way the compiler lays out the bitfields is not defined, this has to be fixed.

Reply:
@lgirdwood I think such change should be made in the PR introducing driver-side xrun handling, since it's a required for this to work anyway.

Reply:
@slawblauciak yes that sounds like a good idea, thanks for the suggestion. we've already removed enums, let's remove implicit bitfields as well from the IPC.
Using definitions as below would be quite nice. 
````c
#define IPC_INFO_BUILD BIT(0)
#define IPC_INFO_LOCKS BIT(1)
#define IPC_INFO_LOCKS_VERBOSE BIT(2)
#define IPC_INFO_GDB BIT(3)
#define IPC_INFO_NO_XRUN_RECOVERY BIT(4)
````

Reply:
@plbossart I've made the PRs for this change
https://github.com/thesofproject/sof/pull/1418
https://github.com/thesofproject/linux/pull/937